### PR TITLE
feat: add documenten api formio component

### PR DESCRIPTION
**Pull request by mritense**
Component is not yet functional, waiting for BE endpoints. The component can still be merged and tested. Closes https://ritense.tpondemand.com/entity/37222-fe-create-new-documenten-api-formio

### CODE REVIEW

File: projects/valtimo/components/src/lib/components/form-io/documenten-api-uploader/documenten-api-uploader.component.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  }
+
+  @Input()
+  public set value(value: Array<ResourceFile>) {
+    if (Array.isArray(value)) {
+      this._value = value;
+    }
+  }
+
+  downloadFile(resourceFile: ResourceFile) {
```
Comment (maisaritense) [inline projects/valtimo/components/src/lib/components/form-io/documenten-api-uploader/documenten-api-uploader.component.ts:85]: missing return type

File: projects/valtimo/components/src/lib/components/form-io/documenten-api-uploader/documenten-api-uploader.component.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    this.uploadProviderService
+      .getResource(resourceFile.data.resourceId)
+      .subscribe((resource: ResourceDto) => {
+        this.downloadService.downloadFile(resource.url, resource.resource.name);
+      });
+  }
+
+  fileSelected(file: File) {
```
Comment (maisaritense) [inline projects/valtimo/components/src/lib/components/form-io/documenten-api-uploader/documenten-api-uploader.component.ts:93]: missing return type

File: projects/valtimo/components/src/lib/components/form-io/documenten-api-uploader/documenten-api-uploader.component.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+  metadataSet(metadata: DocumentenApiMetadata): void {
+    this.uploading$.next(true);
+    this.hideModal$.next(null);
+
+    this.fileToBeUploaded$
+      .pipe(
+        take(1),
+        tap(file => {
+          console.log('upload', file, metadata);
```
Comment (maisaritense) [inline projects/valtimo/components/src/lib/components/form-io/documenten-api-uploader/documenten-api-uploader.component.ts:114]: check the need for this console.log and maybe remove/replace it

File: projects/valtimo/components/src/lib/components/form-io/documenten-api-uploader/documenten-api-uploader.component.scss [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+.dropzone-container {
+  margin-bottom: 12px;
+}
+
+.file-preview {
+  margin-top: -1px;
+  display: flex;
+  border: 1px solid #dee2e6;
+
```
Comment (maisaritense) [inline projects/valtimo/components/src/lib/components/form-io/documenten-api-uploader/documenten-api-uploader.component.scss:47]: `#dee2e6`
maybe create a variable for this color, to reuse. It is used in several places on the **border** property

Screenshot 2022-09-27 at 09 47 15

File: projects/valtimo/components/src/lib/components/form-io/documenten-api-uploader/documenten-api-uploader.component.scss [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  .file-preview-name {
+    width: 65%;
+  }
+
+  .file-preview-size {
+    width: 35%;
+  }
+
+  .icon {
+    color: #a1a1a1;
```
Comment (maisaritense) [inline projects/valtimo/components/src/lib/components/form-io/documenten-api-uploader/documenten-api-uploader.component.scss:61]: maybe create a variable for this color, to reuse.

File: projects/valtimo/components/src/lib/components/form-io/documenten-api-uploader/documenten-api-uploader.component.ts (hunk 3 above)
Comment (mritense) [reply projects/valtimo/components/src/lib/components/form-io/documenten-api-uploader/documenten-api-uploader.component.ts:114]: Will be removed once BE endpoints are ready

File: projects/valtimo/components/src/lib/components/form-io/documenten-api-uploader/documenten-api-uploader.component.scss (hunk 4 above)
Comment (mritense) [reply projects/valtimo/components/src/lib/components/form-io/documenten-api-uploader/documenten-api-uploader.component.scss:47]: This uses the old design system (Bootstrap/Beagle), we don't use design variables for that

File: projects/valtimo/components/src/lib/components/form-io/documenten-api-uploader/documenten-api-uploader.component.scss (hunk 5 above)
Comment (mritense) [reply projects/valtimo/components/src/lib/components/form-io/documenten-api-uploader/documenten-api-uploader.component.scss:61]: This uses the old design system (Bootstrap/Beagle), we don't use design variables for that

**Review by maisaritense — APPROVED** [review]

**Pull request merged**